hawp/fsl/predict.py

```diff
@@ -0,0 +1,136 @@
+import argparse
+import json
+import random
+from collections import defaultdict
+from pathlib import Path
+
+import numpy as np
+import torch
+from PIL import Image
+from tqdm import tqdm
+from torch.utils.data import DataLoader, IterableDataset, get_worker_info
+
+from hawp.base.utils.logger import setup_logger
+
+from hawp.fsl.config import cfg
+from hawp.fsl.dataset.build import build_transform
+from hawp.fsl.model.build import build_model
+
+
+class ImageList(IterableDataset):
+    def __init__(self, image_paths, transform):
+        super().__init__()
+        self.image_paths = image_paths
+        self.transform = transform
+
+    def __iter__(self):
+        if get_worker_info() is not None:
+            raise RuntimeError("Single worker only.")
+        for image_path in self.image_paths:
+            im = Image.open(image_path)
+            w, h = im.size
+            meta = {
+                "filename": image_path,
+                "height": h,
+                "width": w,
+            }
+            yield self.transform(np.array(im)), meta
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(description='HAWP Testing')
+    parser.add_argument('config', help = 'the path of config file')
+    parser.add_argument("images", nargs="*")
+    parser.add_argument("--ckpt", type=str, required=True)
+
+    parser.add_argument("--j2l", default = None, type = float, help = 'the threshold for junction-line attraction')
+    parser.add_argument("--rscale",default=2, type=int, help='the residual scale')
```

Comment:
Where have you found the default value of this?

Reply:
Except for `images`, the options are copy pasted from the benchmark cli